### PR TITLE
[YUNIKORN-1319] expose reservation info in application rest endpoints

### PR DESCRIPTION
**Pull request by lowc1012**
### What is this PR for?
expose reservation info in application rest endpoints

### What type of PR is it?
* [ ] - Bug Fix
* [x] - Improvement
* [ ] - Feature
* [ ] - Documentation
* [ ] - Hot Fix
* [ ] - Refactoring

### Todos
* [ ] - Task

### What is the Jira issue?
[YUNIKORN-1319](https://issues.apache.org/jira/projects/YUNIKORN/issues/YUNIKORN-1319)

### Screenshots (if appropriate)
img1

### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [x] - It needs documentation.

### CODE REVIEW

File: pkg/webservice/handlers_test.go [hunk 1]
```diff
@@ -1068,6 +1068,12 @@ func TestGetQueueApplicationsHandler(t *testing.T) {
 	assert.NilError(t, err, "failed to unmarshal applications dao response from response body: %s", string(resp.outputBytes))
 	assert.Equal(t, len(appsDao), 2)
 
+	if !appsDao[0].HasReserved {
+		assert.Equal(t, len(appsDao[0].Reservations), 0)
+	} else {
+		assert.Check(t, len(appsDao[0].Reservations) > 0, "Get wrong reservation info from apps dao")
```
Comment (manirajv06) [inline pkg/webservice/handlers_test.go:1074]: Can we change the message to "reservation count should have been more than zero" or something like that?

**Review by manirajv06 — APPROVED** [review]
Overall looks good.

+1 pending minor message change

**Review by HuangTing-Yao — APPROVED** [review]
LGTM

File: pkg/webservice/handlers_test.go (hunk 1 above)
Comment (lowc1012) [reply pkg/webservice/handlers_test.go]: Thanks! It makes sense.

**Pull request closed without merging**